pkg/archive/copy.go

```diff
@@ -218,11 +252,11 @@ func CopyInfoDestinationPath(path string) (info CopyInfo, err error) {
 		// Ensure destination parent dir exists.
 		dstParent, _ := SplitPathDirEntry(path)
 
-		parentDirStat, err := os.Lstat(dstParent)
```

Comment:
@jlhawn thanks for the suggestion!  You are correct, os.Stat follows the link.  I updated the PR with the changes and added a test case for symlink -> symlink.